### PR TITLE
Update Teams connector send function parametrizing timeout variables

### PR DESCRIPTION
**Pull request by mpereyra-mi**
Hello, thanks for you project it is being useful for me, I am getting timeout issues from time to time and maybe parametrizing those values will give us more flexibility trying to avoid this issue

### CODE REVIEW

**Comment by sebbmeyer** [comment]
That's a good solution, the issue was mentioned here #15 and I thought maybe to extract them into a separate function to make easier to override the curl options.

Something like this:

```php
class TeamsConnector {

    // ...

   public function send(TeamsConnectorInterface $card) {
        $json = json_encode($card->getMessage());

        $ch = curl_init($this->webhookUrl);
        $this->setCurlOptions($ch);
        // ...
   }
   
   public function setCurlOptions($ch) {
        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
        curl_setopt($ch, CURLOPT_POSTFIELDS, $json);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 3);
        curl_setopt($ch, CURLOPT_HTTPHEADER, [
            'Content-Type: application/json',
            'Content-Length: ' . strlen($json)
        ]);
   }
}
```
And then you could do something like this:
```php
class CustomTeamsConnector extends TeamsConnector {

    public function setCurlOptions($ch) {
        parent::setCurlOptions($ch);
        // override the options you like e.g. timeout
        curl_setopt($ch, CURLOPT_TIMEOUT, 30);
        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 5);
    }

}
´´´

We would be still flexible but we do not have to add every option we like to override as parameter. But I'm open for discussion about this point.

**Comment by mpereyra-mi** [comment]
Yes I understand the solution you are proposing is pretty open to override whatever you want on the curl parameter but in the other hand I am not really seeing why would you change the other parameters, also will be forcing the developer to create his own teamsconnector instead of reusing it quickly from your library

**Comment by sebbmeyer** [comment]
you are probably right, I do not that other parameters will be changed either. But to merge this PR minor changes are necessary because this package is still supporting php 5.4+

File: src/TeamsConnector.php [hunk 1]
```diff
@@ -18,18 +18,20 @@ public function __construct($webhookUrl)
      * Sends card message as POST request
      *
      * @param  TeamsConnectorInterface $card
+     * @param  int $curlOptTimeout by default = 10
+     * @param  int $curlOptConnectTimeout by default = 3
      * @throws Exception
      */
-    public function send(TeamsConnectorInterface $card)
+    public function send(TeamsConnectorInterface $card, int $curlOptTimeout = 10, int $curlOptConnectTimeout = 3)
```
Comment (sebbmeyer) [inline src/TeamsConnector.php]: the type hinting for int is only supported by PHP7.0+ so we need to remove them for now

**Review by sebbmeyer — CHANGES_REQUESTED** [review]

File: src/TeamsConnector.php (hunk 1 above)
Comment (mpereyra-mi) [reply src/TeamsConnector.php]: I have updated the code

**Review by sebbmeyer — APPROVED** [review]

**Pull request merged**